### PR TITLE
#1324: Handle end query param for youtube urls

### PR DESCRIPTION
**Pull request by jnatten**
connects to NDLANO/Issues#1324

### CODE REVIEW

File: src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala [hunk 1]
```diff
@@ -22,7 +22,7 @@ case class OEmbedProvider(providerName: String,
   }
 
   def hostMatches(url: String): Boolean = {
-    Url.parse(url).hostOption == Url.parse(providerUrl).hostOption
+    url.hostOption.exists(providerUrl.hostOption.contains)
```
Comment (gunnarvelle) [inline src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala:25]: Er ikkje url en String? Har String metoden hostOption?

File: src/test/scala/no/ndla/oembedproxy/service/OEmbedConverterServiceTest.scala [hunk 2]
```diff
@@ -39,21 +41,58 @@ class OEmbedConverterServiceTest extends UnitSuite with TestEnvironment {
...
+      .html should equal(expectedResultT)
     OEmbedConverterService
       .addYoutubeTimestampIfdefinedInRequest(requestUrlWithtoutTimestamp, oembed)
       .html should equal(oembed.html)
+    OEmbedConverterService
+      .addYoutubeTimestampIfdefinedInRequest(requestUrlWithEnd, oembed)
+      .html should equal(expectedResultStartEnd)
+  }
+
+  test("That rel=0 also is added to youtube irl if defined in request") {
```
Comment (gunnarvelle) [inline src/test/scala/no/ndla/oembedproxy/service/OEmbedConverterServiceTest.scala]: irl -> url

File: src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala (hunk 1 above)
Comment (jnatten) [reply src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala:25]: url er en string, men `scala-uri` dsl har noe implicit magic :)

File: src/test/scala/no/ndla/oembedproxy/service/OEmbedConverterServiceTest.scala (hunk 2 above)
Comment (jnatten) [reply src/test/scala/no/ndla/oembedproxy/service/OEmbedConverterServiceTest.scala]: will fix :)

File: src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala (hunk 1 above)
Comment (gunnarvelle) [reply src/main/scala/no/ndla/oembedproxy/model/OEmbedProvider.scala:25]: Oooh. Magic! 🎩

**Review by gunnarvelle — APPROVED** [review]

**Review by narhen — APPROVED** [review]

**Pull request merged**